.changes/v3.8.0/899-features.md

```diff
@@ -0,0 +1,2 @@
+* **New Resource:** `vcd_catalog_vapp_template` to manage the upload and usage of vApp Templates [GH-899]
+* **New Data Source:** `vcd_catalog_vapp_template` to fetch existing vApp Templates [GH-899]
```

Comment:
This is a comment about the changelog in general:

Contrary to the PR description, the changelog is very laconic, given that we are deprecating critical resource. What about expanding it to explain the purpose of the new resources? I.e. how it is different from current catalog item. Also, we need a bullet point in the DEPRECATIONS section, which would explain what we are deprecating and what to use instead.

All in all, we need to make sure users understand this is a big change and that they should start using the new resource :)

Reply:
We had a discussion and agreed to defer the deprecation until we have more data. Specifically, we need to:
* merge this PR
* finish PR #901 about VM refactoring
* finish PR #916 about subscribed catalogs, where we may decide that catalog items may still be useful.

Reply:
My fault, while I adjusted the code to what we discussed some time ago, I left the PR description untouched so it had misleading assertions about the deprecation messages that are no longer present in these code changes.

I've updated the PR description so hopefully everything is clearer now.